File: main/jump-game/jump-game.py
class Solution:
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        j = len(nums) - 1
        while j != 0:
            for i in range(j - 1, -2, -1):
                if i + nums[i] >= j:
                    break
            if i == -1:
                return False
            j = i

        return True

File: main/jump-game/jump-game.c
bool canJump(int *const a, int j) {
    --j;

    while (j != 0) {
        int i = j - 1;
        while (i != -1 && i + a[i] < j) --i;
        if (i == -1) return false;
        j = i;
    }

    return true;
}
